ReactWindows/ReactNative.Net46/Views/Scroll/ReactScrollViewManager.cs

```diff
@@ -490,8 +492,96 @@ private static DependencyObject EnsureChild(ScrollViewer view)
+        /// </summary>
+        private static readonly Dictionary<int, CancellationTokenSource> _scollViewCancelMap = new Dictionary<int, CancellationTokenSource>();
+
+        /// <summary>
+        /// ScrollViewer does not support ScrollToSomeWhere animated, this is custom animated work.
+        /// </summary>
+        /// <param name="weakScrollView">WeakRefernce ScrollViewer</param>
+        /// <param name="x">Scroll to x</param>
+        /// <param name="y">Scroll to y</param>
+        /// <returns>Async task once scrolling is done</returns>
+        private static async Task ScrollToAnimated(WeakReference<ScrollViewer> weakScrollView, double x, double y)
+        {
+            double currentScrollOffsetX = 0;
+            double currentScrollOffsetY = 0;
+
+            // Will do animation in this period
+            const int ANIMATED_TIME_INTERVAL = 300;
+
+            // Progressively scroll time unit
+            const int ANIMATED_TIME_UNIT = 100;
+
+            // Total times to do progressive scroll
+            const int ANIMATED_TIME_SLOT = ANIMATED_TIME_INTERVAL / ANIMATED_TIME_UNIT;
+
+            // Threshold to check equalness for double value
+            const double DOUBLE_EQUAL_THRESHOLD = 0.001;
+
+            ScrollViewer scrollView;
+            if (weakScrollView.TryGetTarget(out scrollView))
+            {
+                // Check to cancel previous scrolling
+                var hashCode = scrollView.GetHashCode();
+                if (_scollViewCancelMap.ContainsKey(hashCode))
+                {
+                    _scollViewCancelMap[hashCode].Cancel();
+                } else
+                {
+                    _scollViewCancelMap.Add(hashCode, new CancellationTokenSource());
+                }
+
+                // Setup progressive scrolling settings
+                currentScrollOffsetX = scrollView.HorizontalOffset;
+                currentScrollOffsetY = scrollView.VerticalOffset;
+                var biasX = (x - currentScrollOffsetX) / ANIMATED_TIME_SLOT;
+                var biasY = (y - currentScrollOffsetY) / ANIMATED_TIME_SLOT;
+
+                var isHorizontalScroll = Math.Abs(x - currentScrollOffsetX) >= DOUBLE_EQUAL_THRESHOLD;
+                var isVerticalScroll = Math.Abs(y - currentScrollOffsetY) >= DOUBLE_EQUAL_THRESHOLD;
+
+                while ((isHorizontalScroll || isVerticalScroll) && !_scollViewCancelMap[hashCode].IsCancellationRequested)
+                {
+                    if (isHorizontalScroll)
+                    {
+                        scrollView.ScrollToHorizontalOffset(currentScrollOffsetX + biasX);
+                    }
+
+                    if (isVerticalScroll)
+                    {
+                        scrollView.ScrollToVerticalOffset(currentScrollOffsetY + biasY);
+                    }
+
+                    await Task.Delay(TimeSpan.FromMilliseconds(ANIMATED_TIME_UNIT), _scollViewCancelMap[hashCode].Token);
```

Comment:
this line is pretty dense. can you please extract Token out to an explaining variable?